Now these methods add constraints and return those constraints.

nservidio opened the pull request:
This is useful if you need to keep a reference to the constraint so you can animate it later.

nservidio committed:
Now these methods add constraints and return those constraints. This is useful if you need to keep a reference to the constraint so you can animate it later.

loganwright commented on src/UIView/UIView+Constraints.h:
```diff
@@ -1,5 +1,15 @@
+/**
+ *  When a method returns a dictionary, retrieve the desired contraint by using the following keys:
+ */
+NS_ENUM(NSInteger, IPConstraintKey) {
```
I think maybe these should be string keys.  It's a little confusing to have an integer as a subscript in ObjC since it requires wrapping.

loganwright commented on src/UIView/UIView+Constraints.m:
```diff
@@ -2,58 +2,63 @@
+                                                               attribute:NSLayoutAttributeTop
+                                                              multiplier:1.0
+                                                                constant:insets.top];
+        [constraints setObject:top forKey:@(IPConstraintKeyTop)];
```
Use subscripts: `constraints[IPConstraintKeyTop] = top;`

loganwright commented on src/UIView/UIView+Constraints.h:
```diff
@@ -300,20 +310,20 @@
  *  @param view the view to constrain
  */
-- (void)constrainViewToAllEdges:(UIView *)view;
+- (NSDictionary *)constrainViewToAllEdges:(UIView *)view;
```
I think it'd be nice if these were supported for Swift. Add this to the top of the file:

``` ObjC
typedef NSDictionary<NSString *, NSLayoutConstraint *> ConstraintDictionary;
```

Then use like this:

```
- (ConstraintDictionary *)constrainViewToAllEdges:(UIView *)view;
```

loganwright commented on src/UIView/UIView+Constraints.m:
```diff
@@ -2,58 +2,63 @@
+
     view.translatesAutoresizingMaskIntoConstraints = NO;
-    [self addConstraints:constraints];
+    [self addConstraints:[constraints allValues]];
```
Use `.allValues`

loganwright commented:
Comments => Podspec => :shipit:

brightredchilli commented on src/UIView/UIView+Constraints.h:
```diff
@@ -8,14 +18,14 @@
  *  @param insets the insets to use when positioning the view
  */
-- (void)constrainView:(UIView *)view toInsets:(UIEdgeInsets)insets;
+- (NSDictionary *)constrainView:(UIView *)view toInsets:(UIEdgeInsets)insets;
```
Really neat idea, agree with logan that this should probably be strings though

brightredchilli commented:
:ship:

nservidio committed:
Made dictionaries swift compatible
Used brackets to access dictionary values
Used dot notation for property
Added UIView+Constraints to sample project file
Updated Podspec
